src/borg/archiver.py

```diff
@@ -3220,7 +3220,7 @@ def define_borg_mount(parser):
 
         fs_group = subparser.add_argument_group('Filesystem options')
         fs_group.add_argument('-x', '--one-file-system', dest='one_file_system', action='store_true',
-                              help='stay in the same file system and do not store mount points of other file systems')
+                              help='stay in the same file system and do not store mount points of other file systems. Be aware that this treats every btrfs subvolume as a separate file system and will not back them up! Make doubly sure to check your backup does what you want when using btrfs.')
```

Comment:
this gets rather (too) long for here.

also, i don't want to get into filesystem specifics here (and potentially point out every filesystem that has such specific stuff).

rather than adding it to the (short) help here, one could check whether it is documented how `-x` works (see the check in the code - iirc it is checking for the device number [not] changing) and if that is not documented, add it there.

after talking about how borg -x works, one could also point out that some filesystems trigger this change detection:
- btrfs for subvolumes
- (more to come?)